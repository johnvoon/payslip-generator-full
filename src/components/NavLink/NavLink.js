import React from "react";
import PropTypes from "prop-types";
import { IndexLink, Link, withRouter } from "react-router";
import classNames from "classnames";

const NavLink = ({ router, linkText, slug, index }) => {
  const isActive = router.isActive(slug, true);

  return (
    <li className={classNames({ active: isActive })}>
      {index
        ? <IndexLink to={slug}>
            {linkText}
          </IndexLink>
        : <Link to={slug}>{linkText}</Link>}
    </li>
  );
};

NavLink.propTypes = {
  linkText: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  router: PropTypes.object.isRequired,
  index: PropTypes.bool
};

export default withRouter(NavLink);
