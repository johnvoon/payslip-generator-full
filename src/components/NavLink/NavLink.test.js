import React from "react";
import { IndexLink, Link } from "react-router";
import NavLink from "./NavLink";
import { shallow } from "enzyme";

describe("NavLink", () => {
  let renderedComponent;
  const handleClick = jest.fn();

  beforeEach(() => {
    renderedComponent = shallow(
      <NavLink
        router={{}}
        linkText="Payslip Generator"
        slug="payslip-generator"
      >
        NavLink
      </NavLink>
    );
  });

  it("should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
