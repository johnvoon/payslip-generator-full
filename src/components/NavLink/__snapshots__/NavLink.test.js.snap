exports[`NavLink should match snapshot 1`] = `
<NavLink
  linkText="Payslip Generator"
  router={Object {}}
  slug="payslip-generator">
  NavLink
</NavLink>
`;
